tests/remotes/webdav.py

```diff
@@ -1,44 +1,52 @@
 # pylint:disable=abstract-method
 import os
+import uuid
 from pathlib import Path
+from tempfile import TemporaryDirectory
 from wsgiref.simple_server import make_server
 
 import pytest
 from funcy import first
 from wsgidav.wsgidav_app import WsgiDAVApp
 
 from dvc.path_info import WebDAVURLInfo
-from dvc.utils.fs import makedirs
 from tests.utils.httpd import run_server_on_thread
 
 from .base import Base
 
 AUTH = {"user1": {"password": "password1"}}
+_WEBDAV_ROOT = TemporaryDirectory()
+_WEBDAV_DIR = Path(_WEBDAV_ROOT.name)
 
 
 class Webdav(Base, WebDAVURLInfo):
     @staticmethod
-    def get_url(port):  # pylint: disable=arguments-differ
-        return f"webdav://localhost:{port}"
+    def get_url(port, root_id):  # pylint: disable=arguments-differ
+        return f"webdav://localhost:{port}/{root_id}/"
 
     def mkdir(self, mode=0o777, parents=False, exist_ok=False):
-        assert parents
-        makedirs(self.path, exist_ok=exist_ok)
+        self.dir_path.mkdir(parents=parents, exist_ok=True)
 
     def write_bytes(self, contents):
-        Path(self.path).write_bytes(contents)
+        self.dir_path.write_bytes(contents)
 
     def write_text(self, contents, encoding=None, errors=None):
-        Path(self.path).write_text(contents, encoding=encoding, errors=errors)
+        self.dir_path.write_text(contents, encoding=encoding, errors=errors)
+
+    @property
+    def dir_path(self):
+        return _WEBDAV_DIR / self.path[1:]
 
 
 @pytest.fixture
 def webdav_server(test_config, tmp_path_factory):
     test_config.requires("webdav")
 
     host, port = "localhost", 0
-    directory = os.fspath(tmp_path_factory.mktemp("http"))
-    dirmap = {"/": directory}
+    root_id = str(uuid.uuid4())
+    root_dir = _WEBDAV_DIR / root_id
+    root_dir.mkdir()
+    dirmap = {f"/{root_id}": os.fspath(_WEBDAV_DIR / root_id)}
```

Comment:
Do we actually need the dirmap? The root_id situation is pretty confusing.

Seems like we could just do
```
url = f"webdav://localhost:{webdav_server.port}/{uuid.uuid4()}"
```
in `webdav` fixture and remove `get_url` method (it is a legacy one, we don't really need it, iirc).

Reply:
> Do we actually need the dirmap?

I mean we need the dirmap for `/` to point to _WEBDAV_DIR, but do we need the `root_id` mapping? I might be missing something.

Reply:
> in webdav fixture and remove get_url method (it is a legacy one, we don't really need it, iirc).

This is how we actually isolate each test's own directory

Reply:
Ah, maybe I'm not understanding the `dirmap` correctly. So we have to specify those subdirs in the dirmap or otherwise we won't be able to create/access them?

Reply:
@skshetry Could you take a look at webdav fixture changes, please? We might be missing something.

Reply:
Looks like I did not test the `write_text`/`mkdir` and seems to have been never used before, so didn't notice.

Instead of depending on a certain directory, maybe it'd better to just use a WebDAV client here, similar to what we do for http. What do you guys think? I'll make a PR to update it if we decide so.

Reply:
I guess, in the longer term, we should move away from this `PathInfo` to something that makes it easy to share a state.

Reply:
@skshetry Yeah, that would be pretty good as well. You mean a follow-up PR, right?

Reply:
@efiop, no need to block @isidentical on this, I think.  But, could create a PR now if you like.

Reply:
@skshetry Ok, then let's keep this PR as is for now. Just as long as you are fine with the webdav changes for now. Thank you! :pray: